core/src/main/java/com/sequenceiq/cloudbreak/core/flow/service/AmbariClusterFacade.java

```diff
@@ -254,9 +258,25 @@ public FlowContext upscaleCluster(FlowContext context) throws CloudbreakExceptio
 
     @Override
     public FlowContext runClusterContainers(FlowContext context) throws CloudbreakException {
+        ProvisioningContext provisioningContext = (ProvisioningContext) context;
+        Stack stack = stackService.getById(provisioningContext.getStackId());
+        MDCBuilder.buildMdcContext(stack);
         try {
-            containerRunner.runClusterContainers((ProvisioningContext) context);
-            return context;
+            if (stack.getCluster() != null && stack.getCluster().getStatus().equals(Status.REQUESTED)) {
+                MDCBuilder.buildMdcContext(stack.getCluster());
+                stackUpdater.updateStackStatus(stack.getId(), Status.UPDATE_IN_PROGRESS, "Cluster installation has been started");
```

Comment:
I think it would be better to update the Stack status in an upper layer (like in the facades) and do not leave it to the implementations.